collections/_posts/2024-11-13-AlphaFold2.md

```diff
@@ -0,0 +1,208 @@
+
+The idea behind using the MSA matrix is to analyze the mutations that has occured inside amino acid sequence of proteins that share a common ancestor ad the same function. For instance, intuitively, when comparing the proteins of two species, if two amino acids always mutate at the same time, this can mean that there are really closeby in the 3D structure of the protein. Understanding the correlations in the MSA matrix is thus key to predict the spatial arragement of the target molecule.
+
+The second type of objects processed by the model is the pairwise representation tensor between the elements of an protein sequence. Basically, you can think about it as an embedded version of the pairwise distances between all the atoms of the molecules. At this stage, the model also leverages template protein structure, which are the (embedded) pairwise distances of molecule for which we already know the exact 3D structure; it allows the model to look for similar structures.
+
+
+<div style="text-align:center">
+<img src="/collections/images/AlphaFold/input_embedding.jpg" width=800></div>
+<p style="text-align: center;font-style:italic">Figure 4. AlphaFold input embedding into MSA features and pairwise representation.</p>
+
+&nbsp;
+
+## Evoformer
+
+<div style="text-align:center">
+<img src="/collections/images/AlphaFold/evoformer.jpg" width=800></div>
+<p style="text-align: center;font-style:italic">Figure 5. Structure of AlphaFold's Evoformer.</p>
+
+
+The MSA and pair representations are the two inputs of the first main component of AlphaFold, called **Evoformer** (Figure 5). It is a transformer-based model whose purpose is to update the MSA and pair representations with self and cross attentions. 
+
+One of the key ideas of this transformer is to incorporate clever inductive biases inside the operations performed by the model. As the number of dimensions of the inputs tensors is large -- they can be seen as embedded 2D images --, instead of performing cross-attention on the flattened sequences of tokens, the MSA is modified via row-wise and column-wise self attention operations. Moreover, the MSA only contains feature information, without any information about the structure of the protein. To incorporate such knowledge into the MSA representation, the authors use a modified version of the row-wise self-attenetion where they add inside the softmax operation biases coming from the pair representation tensor (Figure 6). Note that this is only done for the row-wise operations and not the the column-wise ones as it does not really make sense to add this kind of structural information, specific to each protein, to a cross-attention of the information of one amino-acid between severel species (column-wise self-attention is basically the same as Figure 6 without the lower part of the schema).
```

Comment:
```suggestion
One of the key ideas of this transformer is to incorporate clever inductive biases inside the operations performed by the model. As the number of dimensions of the inputs tensors is large -- they can be seen as embedded 2D images --, instead of performing cross-attention on the flattened sequences of tokens, the MSA is modified via row-wise and column-wise self attention operations. Moreover, the MSA only contains feature information, without any information about the structure of the protein. To incorporate such knowledge into the MSA representation, the authors use a modified version of the row-wise self-attention where they add inside the softmax operation biases coming from the pair representation tensor (Figure 6). Note that this is only done for the row-wise operations and not the the column-wise ones as it does not really make sense to add this kind of structural information, specific to each protein, to a cross-attention of the information of one amino-acid between severel species (column-wise self-attention is basically the same as Figure 6 without the lower part of the schema).
```